lib/fs.js

```diff
@@ -2424,7 +2424,7 @@ if (isWindows) {
   // slash.
   const splitRootRe = /^(?:[a-zA-Z]:|[\\/]{2}[^\\/]+[\\/][^\\/]+)?[\\/]*/;
   splitRoot = function splitRoot(str) {
-    return RegExpPrototypeExec(splitRootRe, str)[0];
+    return SideEffectFreeRegExpPrototypeExec(splitRootRe, str)[0];
```

Comment:
```suggestion
    return SideEffectFreeRegExpPrototypeExec(splitRootRe, str)?[0];
```
it’d be a preexisting bug but this exec could be null

Reply:
Unrelated to this PR, I'll leave it as that.